### ISSUE TITLE
[plugin:vanilla-extract] document is not defined

### ISSUE THREAD

**Issue by jesus-hernandezmoreno** [vanilla-extract-css/vanilla-extract#1072]
### Describe the bug

Hello! I've been creating a design system with VE, at first I had evertyhing in one repo, after isolatin the design-system to it's own project I've been encountering the following issue: [plugin:vanilla-extract] document is not defined when I try to use something imported from a css.ts file like sprinkles or a themeContract, I'm assuming the issue happens when everything gets bundled.

This is how I got it setup:

====================
Design System
====================

css.ts

```ts
export { mapResponsiveValue, normalizeResponsiveValue, sprinkles } from './sprinkles.css';
export type {
  OptionalResponsiveObject,
  OptionalResponsiveValue,
  RequiredResponsiveObject,
  RequiredResponsiveValue,
  Sprinkles,
  ComponentProps
} from './sprinkles.css';
export { vars } from './vars.css';
export { themes, colorsContract } from '../themes/contract.css';
export { tokens } from '../tokens';

```

index.ts

```ts
export { Component } from './components/component';

export { sprinkles, vars, themes, tokens, colorsContract } from './css';

export type { Sprinkles, ComponentProps } from './css';

```

NOTE: When I'm developing inside the design system I can refer sprinkles, or colorsContract, and it just works:

CustomComponent.css.ts

```ts
import { style } from '@vanilla-extract/css';
import { sprinkles } from '../../../css';

export const componentStyles = style([{}, sprinkles({display:'flex'})]);

```

vite.config.ts:

```ts
import { defineConfig } from 'vite';
import { vanillaExtractPlugin } from '@vanilla-extract/vite-plugin';
import dts from 'vite-plugin-dts';
import tsconfigPaths from 'vite-tsconfig-paths';

import pkg from './package.json';

export default defineConfig({
  build: {
    lib: {
      entry: 'src/index.ts',
      fileName: 'index',
      formats: ['cjs', 'es']
    },
    rollupOptions: {
      external: Object.keys(pkg.peerDependencies)
    }
  },
  plugins: [
    vanillaExtractPlugin({
      identifiers: 'short'
    }),
    tsconfigPaths(),
    dts({
      exclude: ['src/**/*.docs.mdx', 'src/**/*.snippets.tsx', 'src/**/*.test.ts*'],
      beforeWriteFile: (filePath, content) => ({
        content,
        filePath: filePath.replace('src', '')
      }),
      compilerOptions: {
        baseUrl: './src/',
        emitDeclarationOnly: true,
        noEmit: false
      },
      outputDir: 'dist/types'
    })
  ]
});

```

After getting everything bundled and install the design system on a new project, with vanilla-extract and everything, same versions I get the error

====================
New App
====================

main.tsx
```ts
import 've-design-system/styles';

ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
     <App />
  </React.StrictMode>
);

```

CustomComponent.css.ts

```ts
import { style } from '@vanilla-extract/css';
import { sprinkles } from 've-design-system';

export const componentStyles = style([{}, sprinkles({display:'flex'})]);

```

Hope you can help me, I'm assuming is a configuration issue, because sprinkles, and contracts are created on css.ts files and when I do the export is a simple ts file, not sure.

FYI: I'm developing locally so I still don't have the. desing system opened.

Regards.

### Reproduction

https://github.com/vanilla-extract-css/vanilla-extract/issues/new?assignees=&labels=pending+triage&template=bug_report.yml

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 1.44 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/.nvm/versions/node/v16.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.15.0/bin/yarn
    npm: 8.5.5 - ~/.nvm/versions/node/v16.15.0/bin/npm
    Watchman: 2022.10.17.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 112.0.5615.137
    Firefox: 112.0.1
    Safari: 16.4
  npmPackages:
    @vanilla-extract/css: 1.11.0 => 1.11.0 
    @vanilla-extract/recipes: 0.4.0 => 0.4.0 
    @vanilla-extract/sprinkles: 1.6.0 => 1.6.0 
    @vanilla-extract/vite-plugin: 3.8.0 => 3.8.0 
    vite: ^4.2.1 => 4.3.2
```

### Used Package Manager

npm

### Logs

```shell
document is not defined
10:33:07 AM [vite] Internal server error: document is not defined
  Plugin: vanilla-extract
  File: /Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts
      at Object.<anonymous> (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:36772:19)
      at o (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:33000:25)
      at Object.<anonymous> (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:33471:11)
      at o (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:33000:25)
      at Object.<anonymous> (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:35816:39)
      at o (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:33000:25)
      at Object.<anonymous> (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:37892:19)
      at o (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:33000:25)
      at Object.<anonymous> (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:38719:23)
      at o (/Users/jhernandez/Code/la-cotorrisa/app/src/components/anecdote-card/AnecdoteCard.css.ts:33000:25)
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by mattcompiles** [comment]
Hey Bud, 

Without a small reproduction on the issue it's almost impossible to help. You linked a repro but the link is just pointing to the VE issue template, was this intentional?

**Comment by jesus-hernandezmoreno** [comment]
@mattcompiles Sorry I tried to reproduce it on a CodeSandox but this is more of a configuration issue, which I still don't know how to solve.

I've been doing some testing and the thing is that when I export some css.ts file from the index.ts file it gives me an error, so the question is how can I import css.ts files from a component library ? I want to use:

sprinkles.css.ts
theme.css.ts

I want to do something like:

import {sprinkles, theme} from 'component-library'

**Comment by askoufis** [comment]
This _could_ be a result of how you're exporting styles in your component library. Exporting `sprinkles` and `theme` at the top-level entrypoint of your package can have unintended side-effects if you're exporting components alongside. Could you try moving all shared styles to the `/styles` entrypoint?

**Comment by jesus-hernandezmoreno** [comment]
@askoufis Thanks for the reply, sorry I've been out.

This is my current configuration:

Everything related to css.ts is inside the css folder:

<img width="1889" alt="Screenshot 2023-05-23 at 20 14 50" src="https://github.com/vanilla-extract-css/vanilla-extract/assets/17417706/a24dc9f0-9bc6-4832-85d3-d00035681c92">

This is my vite.config.ts

<img width="1214" alt="Screenshot 2023-05-23 at 20 16 16" src="https://github.com/vanilla-extract-css/vanilla-extract/assets/17417706/66ccf73b-616e-4f8b-ae44-118e2f7c1486">


This is my package.json


<img width="624" alt="Screenshot 2023-05-23 at 20 16 35" src="https://github.com/vanilla-extract-css/vanilla-extract/assets/17417706/a3ea3ad1-561c-4dbc-a6f3-79fa3fb1fdbe">

To be honest I'm not sure how to add it to the style entry, is there a way to  define another entry insised vite ? like havie multiple entries.

**Comment by askoufis** [comment]
This is quite hard to debug from code in the issue and screenshots. A minimal reproduction in a repo somewhere would be much more helpful, but I can give it a shot from what you've provided.

I'm not sure if this will fix it, but you have a `styles` entry already, in your package json `exports`. I can see you're re-exporting the stuff in `css/index.ts` from `index.ts` alongside components. I would suggest _not_ doing that, instead letting consumers do `import { vars } from 'component-library/styles` instead.

Also I'm a bit confused by the package name. Is it called `component-library` or `cbs-design-system`? Again, a reproduction of the issue would be ideal.

**Comment by jesus-hernandezmoreno** [comment]
@askoufis Sorry I know is hard to debug without reproduction and I'm sorry, I can only get the issue when I build the package and try to install it as a dependency, and thanks a lot you actually help me solved the issue!!! :tada:.

You were right:

" Exporting sprinkles and theme at the top-level entrypoint of your package can have unintended side-effects if you're exporting components alongside "

So the solution for my was creating another entry point on the build configuration so I can import all css.tss files from  `component-library/css` and the other components from the `component-library', and it worked.

FYI: Is called `component-library` and `cbs-design-system` I cloned the repo multiple times. 

TLDR:

To been able to import the css.ts files from a component library create one entry for the componentes and all the other stuff and one only for the css.ts files.

<img width="1836" alt="Screenshot 2023-05-24 at 8 43 45" src="https://github.com/vanilla-extract-css/vanilla-extract/assets/17417706/970af887-3cbd-4c70-917f-169dc1d109c5">
